### PR TITLE
chore: add notionID to employee

### PR DESCRIPTION
**Pull request by nguyennh4522**
#### What's this PR do?

- [x] chore: add notionID to employee
- [x] chore: add positions and projects to employee response

#### What are the relevant Git tickets?

https://www.notion.so/dwarves/4d756d46e90240918cd2505f962cacd1?v=d65335d1772f4532ab1bc274a1ae8c76&p=e39a6d00254a4b72a315c5e12ad5ad98&pm=s

#### Screenshots (if appropriate)

image

### CODE REVIEW

File: pkg/model/employees.go [hunk 1]
```diff
@@ -41,17 +41,20 @@ type Employee struct {
...
-	WorkingStatus WorkingStatus
-	JoinedDate    *time.Time
-	LeftDate      *time.Time
-	AccountStatus AccountStatus
+	WorkingStatus     WorkingStatus
+	JoinedDate        *time.Time
+	LeftDate          *time.Time
+	AccountStatus     AccountStatus
+	EmployeePositions []EmployeePosition
+	ProjectMembers    []ProjectMember
```
Comment (namnhce) [inline pkg/model/employees.go:49]: Preload field should be separated to other group

Comment (namnhce) [inline pkg/model/employees.go:49]:
```suggestion
	Projects    []ProjectMember
```

File: pkg/model/projects.go [hunk 2]
```diff
@@ -53,13 +53,13 @@ func (e ProjectStatus) String() string {
...
 	BaseModel
 
-	Name      string
-	Type      ProjectType
-	StartDate *time.Time
-	EndDate   *time.Time
-	Status    ProjectStatus
-	Members   []ProjectMember
-	Heads     []ProjectHead
+	Name      string          `json:"name"`
```
Comment (namnhce) [inline pkg/model/projects.go:56]: Why do we need json here?

**Review by namnhce — CHANGES_REQUESTED** [review]

File: pkg/store/employee/employee.go [hunk 3]
```diff
@@ -30,22 +30,29 @@ func (s *store) OneByTeamEmail(teamEmail string) (*model.Employee, error) {
...
+		query = query.Where("working_status = ?", filter.WorkingStatus)
 	}
-	db = db.Count(&total)
+	query = query.Count(&total)
 
-	if pagination.Page > 1 {
-		db = db.Offset(int((pagination.Page - 1) * pagination.Size))
+	query = query.Order(pagination.Sort)
+	limit, offset := pagination.ToLimitOffset()
+	if pagination.Page > 0 {
```
Comment (namnhce) [inline pkg/store/employee/employee.go:45]: 👍

File: pkg/store/employee/employee.go [hunk 4]
```diff
@@ -30,22 +30,29 @@ func (s *store) OneByTeamEmail(teamEmail string) (*model.Employee, error) {
...
+		query = query.Limit(limit)
 	}
-	db = db.Limit(int(pagination.Size))
-	db = db.Order(pagination.Sort)
 
-	var employees []*model.Employee
-	return employees, total, db.Find(&employees).Error
+	query = query.Preload("ProjectMembers", "deleted_at IS NULL").
```
Comment (namnhce) [inline pkg/store/employee/employee.go:49]: good spot

**Review by namnhce — CHANGES_REQUESTED** [review]

File: pkg/model/projects.go (hunk 2 above)
Comment (nguyennh4522) [reply pkg/model/projects.go:56]: model.Project will return in view `EmployeeData`. Should add `JSON` tag to convert it to camelcase

File: pkg/model/employees.go (hunk 1 above)
Comment (nguyennh4522) [reply pkg/model/employees.go:49]: okay

File: pkg/model/projects.go (hunk 2 above)
Comment (namnhce) [reply pkg/model/projects.go:56]: can we define a new struct in view package?

Comment (nguyennh4522) [reply pkg/model/projects.go:56]: okay

File: pkg/view/employee.go [hunk 5]
```diff
@@ -72,9 +88,14 @@ func ToEmployeeData(employee *model.Employee) *EmployeeData {
 		Gender:        employee.Gender,
 		Horoscope:     employee.Horoscope,
 		DateOfBirth:   employee.DateOfBirth,
+		GithubID:      employee.GithubID,
+		DiscordID:     employee.DiscordID,
+		NotionID:      employee.NotionID,
 		WorkingStatus: employee.WorkingStatus,
 		JoinedDate:    employee.JoinedDate,
 		LeftDate:      employee.LeftDate,
+		Projects:      projects,
```
Comment (namnhce) [inline pkg/view/employee.go:97]: We don't need full project data here. Only need projectID and its name

**Review by namnhce — CHANGES_REQUESTED** [review]

File: pkg/view/employee.go (hunk 5 above)
Comment (nguyennh4522) [reply pkg/view/employee.go:97]: updated

**Review by namnhce — APPROVED** [review]

**Pull request merged**